antarest/matrixstore/service.py

```diff
@@ -455,19 +503,21 @@ def export_task(notifier: TaskUpdateNotifier) -> TaskResult:
 
     def download_matrix(
         self,
-        matrix_id: str,
-        tmp_file: Path,
+        matrix_hash: str,
+        filepath: Path,
         params: RequestParameters,
     ) -> None:
         """
-        Export study output to a zip file.
+        Prepare the matrix download if the user has permissions to do it.
+
         Args:
-            matrix_id: matrix id
-            tmp_file: temporary file
-            params: request parameters
+            matrix_hash: The SHA256 hash of the matrix object to download.
+            filepath: File path of the TSV file to write.
+            params: Request parameters.
         """
         if not params.user:
             raise UserHasNotPermissionError()
-        matrix = self.get(matrix_id)
-        if matrix:
-            write_tsv_matrix_to_file(matrix, str(tmp_file))
+        if matrix := self.get(matrix_hash):
+            array = np.array(matrix.data, dtype=np.float64)
+            # noinspection PyTypeChecker
+            np.savetxt(filepath, array, delimiter="\t", fmt="%.18g")
```

Comment:
> note that default encoding is latin1, not UTF-8, should not be an issue since we have only numbers here, though.

I know, actually this is platform-dependent. On Windows this is Latin-1, on Linux this is UTF8. In the previous implementation it wasn't specified, so it was UTF8 by default. I prefer to use UTF8, or best UTF8-sig 😁

Reply:
> note that default encoding is latin1, not UTF-8, should not be an issue since we have only numbers here, though.

I know, actually this is platform-dependent. On Windows this is Latin-1, on Linux this is UTF8. In the previous implementation it wasn't specified, so it was UTF8 by default. I prefer to use UTF8, or best UTF8-sig 😁